devicetypes/vision-raytseng/vision-4-in-1-motion-sensor.src/vision-4-in-1-motion-sensor.groovy

```diff
@@ -0,0 +1,510 @@
+	parameter.parameter7title = "LED display mode"
+	parameter.parameter7description = ""
+	parameter.parameter7type = "enum"
+	parameter.parameter7options = ["LED off when Temperature report/Motion active",
+								   "LED blink when Temperature report/Motion active",
+								   "LED blink when Motion active/LED off when Temperature report"]
+	parameter.parameter7enumMap = ["LED off when Temperature report/Motion active": 1,
+								   "LED blink when Temperature report/Motion active": 2,
+								   "LED blink when Motion active/LED off when Temperature report": 3]
+	parameter.parameter7default = "LED off when Temperature report/Motion active"
+	parameter.parameter7size = 1
+	
+	parameter.parameter8name = "RetryTimes"
+	parameter.parameter8title = "Motion notification retry times"
+	parameter.parameter8description = ""
+	parameter.parameter8type = "number"
+	parameter.parameter8options = "0..10"
+	parameter.parameter8enumMap = []
+	parameter.parameter8default = 3
+	parameter.parameter8size = 1
+	
+	return parameter."parameter${num}${text}"
+}
+
+def wakeUpIntervalUpdate(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" != value) {
+			state."${getWakeUpInfo("name")}" = value
+			state."${getWakeUpInfo("name")}Refresh" = true
+		}
+	}
+}
+
+def wakeUpIntervalCheck(interval) {
+	def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+	
+	if (value != null) {
+		if (state."${getWakeUpInfo("name")}" == value) {
+			if (state."${getWakeUpInfo("name")}Refresh" == true) {		
+				state."${getWakeUpInfo("name")}Refresh" = false
+			}
+		}
+	}
+}
+
+def getWakeUpInfo(text) {
+	def wakeUp = [:]
+	
+	wakeUp.name = "wakeUpInterval"
+	wakeUp.title = "Wake up interval [Hours]"
+	wakeUp.description = ""
+	wakeUp.type = "number"
+	wakeUp.range = "1..4660"
+	wakeUp.default = 24
+	
+	return wakeUp."${text}"
+}
+
+private getCommandClassVersions() {
+	[
+		0x5E: 2,
+		0x22: 1,
+		0x85: 2,
+		0x59: 3,
+		0x80: 1,
+		0x70: 2,
+		0x5A: 1,
+		0x7A: 5,
+		0x87: 3,
+		0x72: 2,
+		0x8E: 3,
+		0X31: 5,
+		0x71: 3,
+		0x73: 1,
+		0x98: 1,
+		0x9F: 1,
+		0x6C: 1,
+		0x55: 2,
+		0x86: 3,
+		0x84: 2 
+	]
+}
+
+def parse(String description) {
+	def result = []
+	def cmd = zwave.parse(description, commandClassVersions)
+	
+	if (cmd) {
+		result += zwaveEvent(cmd)
+	}
+	else {
+		logDebug "Unable to parse description: ${description}"
+	}
+	
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def cmds = []
+	
+	cmds += configure()
+	if (cmds) {
+		cmds << "delay 5000"
+	}
+	cmds << zwave.wakeUpV2.wakeUpNoMoreInformation().format()
+	
+	return cmds ? response(cmds) : []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	if (cmd.nodeid == zwaveHubNodeId) {
+		def interval = [:]
+		interval."${getWakeUpInfo("name")}" = second2Hour(cmd.seconds)
```

Comment:
Since second2Hour() is used only once, can You just inline this code here ?

Reply:
Okay, I fixed it.